File: hello_world/hello_world.js
#!/usr/bin/env node
const util = require('util');
let str = !process.argv[2] ? "Hello World Nodejs" : process.argv[2];
let starts = 0
let ends   = str.length;
while (true) {
  process.stdout.write(util.format('%s %s \r', str.substring(starts, ends), str.substring(0,starts)));
  starts =  starts > ends ? 0 : starts+1;
  Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, 500); // msleep(500);
}

function msleep(n) {
  Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, n);
}

File: hello_world/HelloWorld.java
public class HelloWorld {
  public static void main(String[] args) {
    String str = args.length == 0 ? "Hello World Java" : args[0];
    int starts = 0;
    int ends = str.length();
    while (true) {
      try {
        System.out.print(str.substring(starts, ends)+" "+str.substring(0, starts)+"\r");
        Thread.sleep(500);
        starts = starts == ends ? 0 : starts+1;
      }
      catch (Exception e) {
        break;
      }
    }
  }
}
